### PR TITLE
Generative VQ-VAE + PixelCNN for the ADNI Brain Dataset - Adrian Rahul Kamal Rajkamal 45811935

### PR DESCRIPTION
**Pull request by arkr01**
Hi,

This pull request contains my implementation of Task 8 (i.e. A generative model of the ADNI brain dataset, using a VQ-VAE with a PixelCNN Prior).

The mean SSIM on the test set, of the VQ-VAE reconstructions, is 0.9162842.

This pull request contains the following:

    modules.py: The source code for all model components (i.e. all models and layers for the VQ-VAE and PixelCNN).
    dataset.py: The source code for loading and pre-processing all data.
    train.py: The source code for training/validating/testing/saving the VQ-VAE and PixelCNN. Training/validation losses are also plotted and saved here.
    predict.py: The source code for loading the trained VQ-VAE and PixelCNN models and visualising example VQ-VAE reconstructions, as well as generating new brain images
    README.md: Project Documentation
    ReducedResults/: Directory containing a small subset of all loss plots/values, reconstructions, SSIM values, etc. generated/calculated, to include in the README

For further details, please refer to the README.

Thanks,
Adrian Rahul Kamal Rajkamal
45811935

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
- No pixel CNN results in readme
- Reconstruction OK
- Image generation seems incomplete, model is there but no code for generation.

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem (no generation code) -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage (no generation output) -1
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
